src/create-with-theme.js

```
@@ -0,0 +1,39 @@
+const React = require('react');
+const PropTypes = require('prop-types');
+const channel = require('./channel');
+
+function createWithTheme(CHANNEL = channel) {
+  return Component =>
+    class withTheme extends React.Component {
+      static contextTypes = {
+        [CHANNEL]: PropTypes.func,
+      };
+
+      componentWillMount() {
+        if (!this.context[CHANNEL]) {
+          throw new Error(
```

i dont know, _fail fast_ seems like a good approach here